### PR TITLE
Implemented transformation of old data format to new

### PR DESCRIPTION
**Pull request by tim-boerner**
Added XSLT to transform metadata Files
Implemented writing of metadata Files

### CODE REVIEW

File: Kitodo-DataEditor/src/main/java/org/kitodo/dataeditor/MetsKitodoReader.java [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+    /**
+     * Reads an old mets-mods-goobi formated xml file by transforming it to the
+     * current used mets-kitodo format.
+     * 
+     * @param xmlFile
+     *            The file as URI object.
+     * @return The Mets object in mets-kitodo format.
+     */
+    static Mets readUriToMetsFromOldFormat(URI xmlFile) throws TransformerException, IOException, JAXBException {
+        URI xslFile = URI.create("./src/main/resources/xslt/MetsModsGoobi_to_MetsKitodo.xsl");
```
Comment (henning-gerhardt) [inline Kitodo-DataEditor/src/main/java/org/kitodo/dataeditor/MetsKitodoReader.java]: Is this hard coded path indented? This will probably not work outside a development system. This mentioned file should be searched in and used from `xslt` folder of the Kitodo.Production installation.

**Comment by tim-boerner** [comment]
Requested changes done

File: Kitodo-DataEditor/src/main/java/org/kitodo/dataeditor/MetsKitodoReader.java [hunk 2]
```diff
@@ -12,6 +12,7 @@
 package org.kitodo.dataeditor;
 
 import java.io.IOException;
+import java.io.InputStream;
```
Comment (henning-gerhardt) [inline Kitodo-DataEditor/src/main/java/org/kitodo/dataeditor/MetsKitodoReader.java]: Why did you add this import statement if it not used?

File: Kitodo-DataEditor/src/main/java/org/kitodo/dataeditor/MetsKitodoReader.java [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+import java.io.StringReader;
+import java.net.URI;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+
+import javax.xml.bind.JAXBContext;
+import javax.xml.bind.JAXBException;
+import javax.xml.bind.Unmarshaller;
+import javax.xml.transform.TransformerException;
+import javax.xml.transform.stream.StreamSource;
```
Comment (henning-gerhardt) [inline Kitodo-DataEditor/src/main/java/org/kitodo/dataeditor/MetsKitodoReader.java]: This import statement is not used.

File: Kitodo-DataEditor/src/main/java/org/kitodo/dataeditor/VersionFinder.java [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+                Attributes mainAttributes = manifest.getMainAttributes();
+                String implementationTitle = mainAttributes.getValue("Implementation-Title");
+                if (implementationTitle != null && implementationTitle.equals(applicationName)) {
+                    String implementationVersion = mainAttributes.getValue("Implementation-Version");
+                    String buildTime = mainAttributes.getValue("Implementation-Build-Date");
+                    return implementationTitle + " - " + implementationVersion + " (" + buildTime + ")";
+                }
+            }
+        }
+        logger.error("Error at writing version info in header of mets file!");
```
Comment (henning-gerhardt) [inline Kitodo-DataEditor/src/main/java/org/kitodo/dataeditor/VersionFinder.java]: This log comment is misleading as version information is not found and not written.

File: Kitodo/src/main/java/org/kitodo/services/dataeditor/DataEditorService.java [hunk 5]
```diff
@@ -15,29 +15,43 @@
 
 import java.io.IOException;
 import java.net.URI;
+import java.nio.file.Paths;
 
+import org.apache.logging.log4j.LogManager;
```
Comment (henning-gerhardt) [inline Kitodo/src/main/java/org/kitodo/services/dataeditor/DataEditorService.java]: Not used import statement.

File: Kitodo/src/main/java/org/kitodo/services/dataeditor/DataEditorService.java [hunk 6]
```diff
@@ -15,29 +15,43 @@
...
+import org.apache.logging.log4j.Logger;
```
Comment (henning-gerhardt) [inline Kitodo/src/main/java/org/kitodo/services/dataeditor/DataEditorService.java]: Not used import statement.

File: Kitodo/src/main/java/org/kitodo/services/dataeditor/DataEditorService.java [hunk 7]
```diff
@@ -15,29 +15,43 @@
...
 import org.kitodo.api.dataeditor.DataEditorInterface;
 import org.kitodo.serviceloader.KitodoServiceLoader;
+import org.kitodo.services.ServiceManager;
```
Comment (henning-gerhardt) [inline Kitodo/src/main/java/org/kitodo/services/dataeditor/DataEditorService.java]: Not used import statement.

**Comment by tim-boerner** [comment]
Requested changes done

File: Kitodo-DataEditor/src/test/java/org/kitodo/dataeditor/MetsKitodoValidatorTest.java [hunk 8]
```diff
@@ -0,0 +1,44 @@
...
+import org.junit.Assert;
+import org.junit.Test;
+import org.kitodo.dataformat.metskitodo.Mets;
+
+public class MetsKitodoValidatorTest {
+
+    @Test
+    public void shouldCheckValidMetsObject() throws JAXBException, IOException {
+        Mets mets = MetsKitodoReader.readUriToMets(Paths.get("./src/test/resources/testmeta.xml").toUri());
+        Assert.assertTrue("Result of validation of Mets object was not true!",MetsKitodoValidator.checkMetsKitodoFormatOfMets(mets));
```
Comment (henning-gerhardt) [inline Kitodo-DataEditor/src/test/java/org/kitodo/dataeditor/MetsKitodoValidatorTest.java]: Missing white space between method parameters.

File: Kitodo-DataEditor/src/test/java/org/kitodo/dataeditor/MetsKitodoValidatorTest.java [hunk 9]
```diff
@@ -0,0 +1,44 @@
...
+    }
+
+    @Test
+    public void shouldCheckOldFormatMetsObject() throws JAXBException, IOException {
+        Mets mets = MetsKitodoReader.readUriToMets(Paths.get("./src/test/resources/testmetaOldFormat.xml").toUri());
+        Assert.assertFalse("Result of validation of Mets object was not false!",MetsKitodoValidator.checkMetsKitodoFormatOfMets(mets));
```
Comment (henning-gerhardt) [inline Kitodo-DataEditor/src/test/java/org/kitodo/dataeditor/MetsKitodoValidatorTest.java]: Missing white space between method parameters.

File: Kitodo-DataEditor/src/test/java/org/kitodo/dataeditor/MetsKitodoValidatorTest.java [hunk 10]
```diff
@@ -0,0 +1,44 @@
...
+    }
+
+    @Test
+    public void shouldMetsContainsMetadataAtMdSecIndex() throws JAXBException, IOException {
+        Mets mets = MetsKitodoReader.readUriToMets(Paths.get("./src/test/resources/testmeta.xml").toUri());
+        Assert.assertTrue("Result of checking if mets contains metadata at dmdSec index was wrong!",MetsKitodoValidator.metsContainsMetadataAtDmdSecIndex(mets,2));
```
Comment (henning-gerhardt) [inline Kitodo-DataEditor/src/test/java/org/kitodo/dataeditor/MetsKitodoValidatorTest.java]: Missing white space between method parameters.

File: Kitodo-DataEditor/src/test/java/org/kitodo/dataeditor/MetsKitodoValidatorTest.java [hunk 11]
```diff
@@ -0,0 +1,44 @@
...
+        Assert.assertFalse("Result of checking if mets contains metadata at dmdSec index which does not exist was wrong!",MetsKitodoValidator.metsContainsMetadataAtDmdSecIndex(mets,3));
```
Comment (henning-gerhardt) [inline Kitodo-DataEditor/src/test/java/org/kitodo/dataeditor/MetsKitodoValidatorTest.java]: Missing white space between method parameters.

File: Kitodo-DataEditor/src/test/java/org/kitodo/dataeditor/MetsKitodoPrefixMapperTest.java [hunk 12]
```diff
@@ -0,0 +1,31 @@
...
+public class MetsKitodoPrefixMapperTest {
+    MetsKitodoPrefixMapper mapper = new MetsKitodoPrefixMapper();
+
+    @Test
+    public void shouldReturnPrefix() {
+        String preferredKitodoPrefix = mapper.getPreferredPrefix("http://meta.kitodo.org/v1/", null, true);
+        String preferredMetsPrefix = mapper.getPreferredPrefix("http://www.loc.gov/METS/", null, true);
+        String preferredXlinkPrefix = mapper.getPreferredPrefix("http://www.w3.org/1999/xlink", null, true);
+        String notExistingPrefix = mapper.getPreferredPrefix("http://not.existing", "return this", true);
+        Assert.assertEquals("Prefix mapper return the wrong prefix for kitodo uri","kitodo",preferredKitodoPrefix);
```
Comment (henning-gerhardt) [inline Kitodo-DataEditor/src/test/java/org/kitodo/dataeditor/MetsKitodoPrefixMapperTest.java]: Missing white space between method parameters.

File: Kitodo-DataEditor/src/test/java/org/kitodo/dataeditor/MetsKitodoPrefixMapperTest.java [hunk 13]
```diff
@@ -0,0 +1,31 @@
...
+        Assert.assertEquals("Prefix mapper return the wrong prefix for mets uri","mets",preferredMetsPrefix);
```
Comment (henning-gerhardt) [inline Kitodo-DataEditor/src/test/java/org/kitodo/dataeditor/MetsKitodoPrefixMapperTest.java]: Missing white space between method parameters.

File: Kitodo-DataEditor/src/test/java/org/kitodo/dataeditor/MetsKitodoPrefixMapperTest.java [hunk 14]
```diff
@@ -0,0 +1,31 @@
...
+        Assert.assertEquals("Prefix mapper return the wrong prefix for xlink uri","xlink",preferredXlinkPrefix);
```
Comment (henning-gerhardt) [inline Kitodo-DataEditor/src/test/java/org/kitodo/dataeditor/MetsKitodoPrefixMapperTest.java]: Missing white space between method parameters.

File: Kitodo-DataEditor/src/test/java/org/kitodo/dataeditor/MetsKitodoPrefixMapperTest.java [hunk 15]
```diff
@@ -0,0 +1,31 @@
...
+        Assert.assertEquals("Prefix mapper return the wrong prefix for not existing uri","return this",notExistingPrefix);
```
Comment (henning-gerhardt) [inline Kitodo-DataEditor/src/test/java/org/kitodo/dataeditor/MetsKitodoPrefixMapperTest.java]: Missing white space between method parameters.

**Comment by tim-boerner** [comment]
dooone
